engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -217,6 +217,7 @@
   protected String shutdownDiagnosticsMessage = "";
   private long lastResourceRequest = 0;
   private final Map<String, StreamingContainerAgent> containers = new ConcurrentHashMap<>();
+  private final Set<String> containerHeartBeatMissSet = Collections.newSetFromMap(new ConcurrentHashMap<String, Boolean>());
```

Comment:
What is the significance of the Set or a Collection of containers that missed the heartbeat? Can you introduce boolean on StreamingContainerAgent that identifies containers that missed the heartbeat? Another option will be to rely on the lastHeartBeatMillis. Update it to -1 (check what will be a good option) when the container misses heartbeat for the first time.